engine/spv_wallet_engine_suite_test.go

```diff
@@ -88,100 +81,51 @@ func (ts *EmbeddedDBTestSuite) TearDownTest() {
+				MaxIdleConnections: 1,
+				MaxOpenConnections: 1,
+				TablePrefix:        tablePrefix,
+			},
+			Shared: true, // mrz: TestTransaction_Save requires this to be true for some reason
+			// I get the error: no such table: _17a1f3e22f2eec56_utxos
+		}))
+	} else if database == datastore.PostgreSQL {
+
+		// Sanity check
+		if ts.PostgresqlServer == nil {
+			return nil, ErrLoadServerFirst
 		}
 
-		// Switch on database types
-		if database == datastore.SQLite {
-			opts = append(opts, WithCustomFeeUnit(mockFeeUnit), WithSQLite(&datastore.SQLiteConfig{
-				CommonConfig: datastore.CommonConfig{
-					MaxConnectionIdleTime: 0,
-					MaxConnectionTime:     0,
-					MaxIdleConnections:    1,
-					MaxOpenConnections:    1,
-					TablePrefix:           tablePrefix,
-				},
-				ExistingConnection: tc.SQLConn,
-			}))
-		} else if database == datastore.PostgreSQL {
-			opts = append(opts, WithSQLConnection(datastore.PostgreSQL, tc.SQLConn, tablePrefix))
-		} else {
-			return nil, ErrDatastoreNotSupported
-		}
+		// Add the new Postgresql connection
+		opts = append(opts, WithSQL(datastore.PostgreSQL, &datastore.SQLConfig{
+			CommonConfig: datastore.CommonConfig{
+				MaxIdleConnections: 1,
+				MaxOpenConnections: 1,
+				TablePrefix:        tablePrefix,
+			},
+			Host:     postgresqlTestHost,
+			Name:     postgresqlTestName,
+			User:     postgresqlTestUser,
+			Password: postgresTestPassword,
+			Port:     fmt.Sprintf("%d", postgresqlTestPort),
+		}))
 
 	} else {
-		// Load the in-memory version of the database
-		if database == datastore.SQLite {
-			opts = append(opts, WithSQLite(&datastore.SQLiteConfig{
-				CommonConfig: datastore.CommonConfig{
-					MaxIdleConnections: 1,
-					MaxOpenConnections: 1,
-					TablePrefix:        tablePrefix,
-				},
-				Shared: true, // mrz: TestTransaction_Save requires this to be true for some reason
-				// I get the error: no such table: _17a1f3e22f2eec56_utxos
-			}))
-		} else if database == datastore.PostgreSQL {
-
-			// Sanity check
-			if ts.PostgresqlServer == nil {
-				return nil, ErrLoadServerFirst
-			}
-
-			// Add the new Postgresql connection
-			opts = append(opts, WithSQL(datastore.PostgreSQL, &datastore.SQLConfig{
-				CommonConfig: datastore.CommonConfig{
-					MaxIdleConnections: 1,
-					MaxOpenConnections: 1,
-					TablePrefix:        tablePrefix,
-				},
-				Host:     postgresqlTestHost,
-				Name:     postgresqlTestName,
-				User:     postgresqlTestUser,
-				Password: postgresTestPassword,
-				Port:     fmt.Sprintf("%d", postgresqlTestPort),
-			}))
-
-		} else {
-			return nil, ErrDatastoreNotSupported
-		}
-	}
-
-	// Custom for SQLite and Mocking (cannot ignore the version check that GORM does)
-	if mockDB && database == datastore.SQLite {
-		tc.MockSQLDB.ExpectQuery(
-			"select sqlite_version()",
-		).WillReturnRows(tc.MockSQLDB.NewRows([]string{"version"}).FromCSVString(sqliteTestVersion))
-	}
-
-	// Are we mocking redis?
-	if mockRedis {
```

Comment:
🤔 it wasn't used anywhere? or you changed the tests that was using it?

Reply:
It was used in the tests what I'm removing :P (The ones that use sqlmock)